### DISCUSSION TITLE
a wonderful idea  for simplicity from midway hooks

### DISCUSSION THREAD

**Discussion by scil** [blitz-js/blitz#1950] [Ideas]
[midway](https://github.com/midwayjs/hooks)  "Using React Hooks to develop the back-end ". Like blitz, it uses http request to replace back-end functions which need context info like req, res.

blitz use `useQuery` `useMutation` to make context available, while midway uses a more simple method which provide context not by functions arguments , but by a react-hook-like approach.

Pros: more simple and user-friendly; no zod like tools to ensure type safety, plain ts is enouth.

**Comment by flybayer** [comment]
Thanks @scil! That's an interesting project I haven't seen before.

Not sure if we can use that approach though. Zod is needed primary for runtime validation. Types are nice secondary bonus.

`useContext` on the backend is definitely cool, but it makes things more complex if you want to call that function directly from `blitz console` or some other backend code. Whereas now if you need `ctx`, you explicitly pass it via the second argument. If that second argument is replaced with `useContext`, how do you pass context?

**Reply by scil** [reply]
Thanks for your opinon.  Let me spend more time contemplating it.
